tests/bundle/test_config_parser.py

```
@@ -388,5 +390,89 @@ def test_load_configs(
+    def test_delete_write_through(self):
+        del self.parser.A.B["C"]
+        self.assertNotIn("C", self.parser.get_parsed_content("A::B"))
+        del self.parser.training.trainer
+        self.assertNotIn("trainer", self.parser.get_parsed_content("training"))
+
+    def test_copy_and_pickle_yield_raw_container(self):
+        # proxies copy/pickle as their underlying container (pre-proxy behaviour).
+        a = self.parser.A
+        self.assertEqual(copy.copy(a), {"B": {"C": 1, "D": [10, 20]}})
+        self.assertEqual(copy.deepcopy(a), {"B": {"C": 1, "D": [10, 20]}})
+        self.assertEqual(pickle.loads(pickle.dumps(a)), {"B": {"C": 1, "D": [10, 20]}})
```

Ruff (0.15.13)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)